ggshield/scan/repo.py

```diff
@@ -75,13 +78,55 @@ def scan_commit(
     except Exception as exc:
         results = Results.from_exception(exc)
 
-    return ScanCollection(
-        commit.sha or "unknown",
-        type="commit",
-        results=results,
-        optional_header=commit.optional_header,
-        extra_info=commit.info._asdict(),
-    )
+    scans = []
+    for commit in commits:
+        concerned_results = [
+            res
+            for res in results.results
+            if any(
+                res.content == file.document and res.filename == file.filename
+                for file in commit.files
+            )
+        ]
+        scans.append(
+            ScanCollection(
+                commit.sha or "unknown",
+                type="commit",
+                results=Results(
+                    results=concerned_results,
+                    errors=results.errors,
+                ),
+                optional_header=commit.optional_header,
+                extra_info=commit.info._asdict(),
+            )
+        )
+
+    return ScanCollection(id=scan_context.command_id, type="commit-ranges", scans=scans)
+
+
+def get_commits_by_batch(
+    commit_list: List[Commit],
+    batch_max_size: int = MULTI_DOCUMENT_LIMIT,
+) -> Iterator[List[Commit]]:
+    """
+    Given a list of commit shas yield the commit files
+    by biggest batches possible of length at most MULTI_DOCUMENT_LIMIT
+    """
+    commits_with_num_files = ((commit, len(commit.files)) for commit in commit_list)
+    current_count = 0
+    batch = []
+    for commit, num_files in commits_with_num_files:
```

Comment:
I think we can get rid of `commits_with_num_files`:
```suggestion
    current_count = 0
    batch = []
    for commit in commit_list:
        num_files = len(commit.files)
```